googletest/src/gtest.cc

```diff
@@ -4914,6 +4914,16 @@ int UnitTest::Run() {
           0x0,                                    // Clear the following flags:
           _WRITE_ABORT_MSG | _CALL_REPORTFAULT);  // pop-up window, core dump.
 # endif
+
+    // In debug mode, the Windows CRT can crash with an assertion over invalid
+    // input (e.g. passing an invalid file descriptor).  The default handling
+    // for these assertions is to pop up a dialog and wait for user input.
+    // Instead ask the CRT to dump such assertions to stderr non-interactively.
+    if (!IsDebuggerPresent()) {
+      (void)_CrtSetReportMode(_CRT_ASSERT,
```

Comment:
Why does this set only `_CRT_ASSERT` and not `_CRT_WARN` and `_CRT_ERROR` as well? is there a reason why the latter two do not need to also be redirected to stderr like this one?